### PR TITLE
Liquid product with short-term investment horizon

### PR DESCRIPTION
**Pull request by lucasia**
Liquid product with short-term investment horizon

### CODE REVIEW

File: packages/contracts/src/interest/DynamicDualRateContext.sol [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+ */
+contract DynamicDualRateContext is CalcInterestMetadata, IDynamicDualRateContext {
+    using EnumerableMap for EnumerableMap.UintToUintMap;
+    using Math for uint256;
+
+    error DynamicDualRateContext_InvalidPeriodRange(uint256 from, uint256 to);
+    error DynamicDualRateContext_InvalidReducedRatePeriod(uint256 from);
+
+    event DynamicDualRateContext_ReducedRateAdded(uint256 period, uint256 rateScaled, uint256 scale);
+    event DynamicDualRateContext_ReducedRateRemoved(uint256 period, uint256 rateScaled, uint256 scale);
```
Comment (lucasia) [inline packages/contracts/src/interest/DynamicDualRateContext.sol:23]: hi @jplodge-pro - please check yarn lint

./src/interest/DynamicDualRateContext.sol
  22:5  warning  Event name must be in CamelCase                                    event-name-camelcase
  23:5  warning  Event name must be in CamelCase                                    event-name-camelcase

Comment (jplodge-pro) [reply packages/contracts/src/interest/DynamicDualRateContext.sol:23]: Yes, I saw that but thought there were pre-existing cases. It IS there for `error`s so I thought the same of `event`s. Thanks, will fix

File: packages/contracts/src/interest/DynamicDualRateContext.sol [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+     * @param toPeriod The [uint256] period to which to determine the effective Reduced Rates. Must be after
+     *  `fromPeriod`.
+     * @return reducedRatesScaled An array of pairs of `period` to `reducedRateScaled` of the Reduced Rates.
+     */
+    function reducedRatesFor(uint256 fromPeriod, uint256 toPeriod)
+        public
+        view
+        override
+        returns (uint256[][] memory reducedRatesScaled)
+    {
```
[the remark this answers is not in the document]
Comment (jplodge-pro) [reply packages/contracts/src/interest/DynamicDualRateContext.sol:89]: I will apply these changes to the `IDualRateContext` as I think that is more correct.

[the remark this answers is not in the document]
Comment (jplodge-pro) [reply packages/contracts/src/interest/DynamicDualRateContext.sol]: Added `ITripleRateContext` and initial implementation. Can discuss is that is satisfactory.

**Pull request merged**